File: 프로그래머스/2/64065. 튜플/튜플.py
def solution(s):
    answer: list[int] = []
    s = s[2:-2]
    tuple_set: list[str] = s.split("},{")
    result_set: set[int] = set()
    tuple_set.sort(key=len)
    
    for ts in tuple_set:
        numbers: list[str] = ts.split(",")
        for number in numbers:
            num: int = int(number)
            if num not in result_set:
                result_set.add(num)
                answer.append(num)
    
    return answer

File: 프로그래머스/2/64065. 튜플/튜플.java
import java.util.*;

class Solution {
    public List<Integer> solution(String s) {
        List<Integer> answer = new ArrayList<>();
        Set<Integer> set = new HashSet<>();
        String subS = s.substring(2, s.length() - 2);
        String[] subTuple = subS.split("\\},\\{");
        Arrays.sort(subTuple, (a, b) -> a.length() - b.length());
        for (int i = 0; i < subTuple.length; i++) {
            String[] numbers = subTuple[i].split(",");
            for (String number : numbers) {
                int num = Integer.parseInt(number);
                if (!set.contains(num)) {
                    set.add(num);
                    answer.add(num);
                }
            }
        }
            
        
        return answer;
    }
}